front/src/modules/rollingStock/helpers/utils.ts

```diff
@@ -375,3 +469,86 @@ export const splitRollingStockProperties = (
 };
+
+/**
+ * According the conversion factors schema, convert a value if
+ * the factors exist in the schema
+ */
+export const convertUnits = (prevUnit: string, newUnit: string, prevValue: number): number => {
+  if (prevUnit in conversionFactorsSchema && newUnit in conversionFactorsSchema[prevUnit]) {
+    return conversionFactorsSchema[prevUnit][newUnit] * prevValue;
+  }
+  return prevValue;
+};
+
+/**
+ * For the rollingstock resistance (a, b or c), check if its unit
+ * to be converted or the unit to convert to is by ton (i.e. daN/t).
+ *
+ * As the resistance mass unit can only be in ton, return the converted
+ * resistance depending if the rollingstock's current mass is in kg or ton
+ */
+export const handleConverWithMass = (
+  previousUnit: string,
+  newUnit: string,
+  currentMassValue: number,
+  currentMassUnit: string,
+  previousValue: number
+) => {
+  let convertedValue = previousValue;
+  if (newUnit.slice(-1) === 't') {
+    const formatedNewUnit = newUnit.slice(0, -2);
+    convertedValue = convertUnits(previousUnit, formatedNewUnit, convertedValue);
+    if (currentMassUnit === 'kg') {
+      currentMassValue = convertUnits('kg', 't', currentMassValue);
+    }
+    convertedValue /= currentMassValue;
+  }
+  if (previousUnit.slice(-1) === 't') {
+    const formatedPreviousUnit = previousUnit.slice(0, -2);
+    convertedValue = convertUnits(formatedPreviousUnit, newUnit, convertedValue);
+    if (currentMassUnit === 'kg') {
+      currentMassValue = convertUnits('kg', 't', currentMassValue);
+    }
+    convertedValue *= currentMassValue;
+  }
+  return convertedValue;
+};
+
+export const isConversionWithTon = (previousUnit: string, newUnit: string) =>
+  previousUnit !== 't' &&
+  newUnit !== 't' &&
+  (previousUnit.slice(-1) === 't' || newUnit.slice(-1) === 't');
+
+/**
+ * When switching unit for the rollingstock's multi units parameters,
+ * convert the parameter value or its min/max interval into the good unit
+ * depending if there is an unit and if it has changed or not.
+ */
+export const handleUnitValue = (
+  option: InputGroupSNCFValue,
+  param: MultiUnitsParameter,
+  currentMass?: RollingStockParametersValues['mass'],
+  valueType: 'value' | 'min' | 'max' = 'value'
+): number | undefined => {
+  const valueToConvert = valueType === 'value' ? option.value : param[valueType];
+  if (valueToConvert !== '' && valueToConvert !== undefined) {
+    if (option.unit !== param.unit && valueToConvert !== 0) {
+      let result;
+      if (currentMass && isConversionWithTon(param.unit, option.unit)) {
+        result = handleConverWithMass(
+          param.unit,
+          option.unit,
+          currentMass.value,
+          currentMass.unit,
+          +valueToConvert
+        );
+      } else {
+        result = convertUnits(param.unit, option.unit, +valueToConvert);
+      }
+      return result;
+    }
+    return +valueToConvert;
+  }
+  return undefined;
+};
```

Comment:
- Reduce if branches
- adding descriptions for each parameter in the function comments as well.
```suggestion
export const handleUnitValue = (
  option: InputGroupSNCFValue,
  param: MultiUnitsParameter,
  currentMass?: RollingStockParametersValues['mass'],
  valueType: 'value' | 'min' | 'max' = 'value'
): number | undefined => {
  const valueToConvert = valueType === 'value' ? option.value : param[valueType];

  if (valueToConvert === '' || valueToConvert === undefined) return undefined;

  if (option.unit !== param.unit && valueToConvert !== 0) {
    const result =
      currentMass && isConversionWithTon(param.unit, option.unit)
        ? handleConverWithMass(
            param.unit,
            option.unit,
            currentMass.value,
            currentMass.unit,
            +valueToConvert
          )
        : convertUnits(param.unit, option.unit, +valueToConvert);

    return result;
  }
  return +valueToConvert;
};
```

Reply:
done